src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -7322,6 +7279,1402 @@ public Test(int b1, int b2, int b3)
+    {
+    }
+
+    partial void Test3([UnscopedRef] Span<long> a)
+    {
+    }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll);
+
+            comp.VerifyDiagnostics(
+                // (14,18): error CS0758: Both partial method declarations must use a params parameter or neither may use a params parameter
+                //     partial void Test1([UnscopedRef] Span<long> a)
+                Diagnostic(ErrorCode.ERR_PartialMethodParamsDifference, "Test1").WithLocation(14, 18),
+                // (14,25): error CS9063: UnscopedRefAttribute cannot be applied to this parameter because it is unscoped by default.
+                //     partial void Test1([UnscopedRef] Span<long> a)
+                Diagnostic(ErrorCode.ERR_UnscopedRefAttributeUnsupportedTarget, "UnscopedRef").WithLocation(14, 25),
+                // (18,18): error CS0758: Both partial method declarations must use a params parameter or neither may use a params parameter
+                //     partial void Test2([UnscopedRef] Span<long> a)
+                Diagnostic(ErrorCode.ERR_PartialMethodParamsDifference, "Test2").WithLocation(18, 18),
+                // (18,25): error CS9063: UnscopedRefAttribute cannot be applied to this parameter because it is unscoped by default.
+                //     partial void Test2([UnscopedRef] Span<long> a)
+                Diagnostic(ErrorCode.ERR_UnscopedRefAttributeUnsupportedTarget, "UnscopedRef").WithLocation(18, 25),
+                // (22,25): error CS9063: UnscopedRefAttribute cannot be applied to this parameter because it is unscoped by default.
+                //     partial void Test3([UnscopedRef] Span<long> a)
+                Diagnostic(ErrorCode.ERR_UnscopedRefAttributeUnsupportedTarget, "UnscopedRef").WithLocation(22, 25)
+                );
+        }
+
+        [Fact]
+        public void ParameterRefSafetyScope_25_Mismatch_Partial()
+        {
+            var src = @"
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+partial class C1
+{
+    partial void Test1(params Span<long> a);
+    partial void Test2(params scoped Span<long> a);
+    partial void Test3(scoped Span<long> a);
+}
+
+partial class C1
+{
+    partial void Test1([UnscopedRef] params Span<long> a)
+    {
+    }
+
+    partial void Test2([UnscopedRef] params Span<long> a)
+    {
+    }
+
+    partial void Test3([UnscopedRef] params Span<long> a)
+    {
+    }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll);
+
+            comp.VerifyDiagnostics(
+                // (22,18): error CS0758: Both partial method declarations must use a params parameter or neither may use a params parameter
+                //     partial void Test3([UnscopedRef] params Span<long> a)
+                Diagnostic(ErrorCode.ERR_PartialMethodParamsDifference, "Test3").WithLocation(22, 18)
+                );
+        }
```

Comment:
I think we need to modify these mismatch tests, or add more versions. The interesting cases for mismatches are cases where a missed `scoped` on an overridden method could cause a safety issue. For example, in the below code example, `M1` and `M2` have a compile error due to mismatched `scoped`, while `M3` does not. All the examples above are of `M3`, as far as I can tell, so we should have variants that look like `M1` and `M2` that use `params` parameters.

```cs
class Base
{
    public virtual void M1(scoped Span<string> s, ref Span<string> s2) { }
    public virtual Span<string> M2(scoped Span<string> s) => throw new NotImplementedException();
    public virtual void M3(scoped Span<string> s) { }
}

class Derived : Base
{
    // error CS8987: The 'scoped' modifier of parameter 's' doesn't match overridden or implemented member.
    public override void M1(Span<string> s, ref Span<string> s2) => s2 = s;
    // error CS8987: The 'scoped' modifier of parameter 's' doesn't match overridden or implemented member.
    public override Span<string> M1(Span<string> s) => s;
    // Fine because there's no chance of `s` getting exposed.
    public override void M3(Span<string> s) { }
}
```

Reply:
> I think we need to modify these mismatch tests

Thanks. I thought I was missing something because I was not getting any mismatch errors even for non-params scenarios.

Reply:
I updated the tests